core/src/main/java/apoc/trigger/TriggerHandlerNewProcedures.java

```diff
@@ -45,89 +34,84 @@ public static void checkEnabled() {
         }
     }
 
-    public static Map<String, Object> install(String databaseName, String triggerName, String statement, Map<String,Object> selector, Map<String,Object> params) {
-        final HashMap<String, Object> previous = new HashMap<>();
+    public static TriggerInfo install(String databaseName, String triggerName, String statement, Map<String,Object> selector, Map<String,Object> params) {
+        AtomicReference<TriggerInfo> previous = new AtomicReference<>();
 
         withSystemDb(tx -> {
             Node node = Util.mergeNode(tx, SystemLabels.ApocTrigger, null,
                     Pair.of(SystemPropertyKeys.database.name(), databaseName),
                     Pair.of(SystemPropertyKeys.name.name(), triggerName));
             
             // we'll return previous trigger info
-            previous.putAll(toTriggerInfo(node));
+            previous.set( fromNode(node, true) );
             
             node.setProperty(SystemPropertyKeys.statement.name(), statement);
             node.setProperty(SystemPropertyKeys.selector.name(), Util.toJson(selector));
             node.setProperty(SystemPropertyKeys.params.name(), Util.toJson(params));
             node.setProperty(SystemPropertyKeys.paused.name(), false);
 
             setLastUpdate(databaseName, tx);
-            return null;
         });
 
-        return previous;
+        return previous.get();
     }
```

Comment:
Hi Giuseppe, I didn't notice this in the dev PR but could you explain why you think you need an atomic reference here? 

AtomicReference is a concurrent programming construct, but this function is executed in a single thread and has no mutable state which is accessible from another thread. So the way I see it, it shouldn't be needed.

Reply:
It's not really doing any harm, but it's just confusing to see here and makes the reader question what is happening and why it's needed, which doesn't help with maintainability. So if there is no use for it, could we please remove it from here and dev too?

Reply:
Hi @AzuObs, I used an `AtomicReference` just to get rid of the `Variable used in lambda expression should be final or effectively final` compilation's error within the lambda.

I just changed them to a one element final array

Reply:
Apologies, now that you mention, I remember us having a conversation about this before. Thanks for making the change.